### ISSUE TITLE
Bug: no-undef reports on class index signature

### ISSUE THREAD

**Issue by hans000** [typescript-eslint/typescript-eslint#9053]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&showAST=es&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgJIDED2nkG8BQyyIcAthAFzIDOYUoA5ocgNoAeVt9IDAulXBABPfAF98%2BBABs41asiw4CRdpzqN%2ByQSKIlya7k3FA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaWXAE0QDN0pFpp9pJwwBfEFoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYwAviNUA&tokens=false

### Repro Code

```TypeScript
interface IFoo {
  name: string
  [x: string]: any
}

class Foo {
  [x: string]: any
  name: string
}
```

### ESLint Config

```json
{
  "rules": {
    "no-undef": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error

### Actual Result

`'x' is not defined.` on the class
